### ISSUE TITLE
Question

### ISSUE THREAD

**Issue by iDevelopper** [vtourraine/VTAcknowledgementsViewController#71]
Not an issue.

Are there any particular reasons for maintaining the two repos?

https://github.com/vtourraine/VTAcknowledgementsViewController

https://github.com/vtourraine/AcknowList

**Comment by vtourraine** [comment]
Historical reasons, I guess? 😅 

The library started as ObjC. Then I tried to re-implement it in Swift when it was introduced. Both projects are still being used, therefore both are still being maintained. Some people want ObjC only, others Swift only, and there are good reasons for both.

The sane thing would be to retire one of them, and just maintain one. I’ll probably do that at one point.

PS: I’m thinking about adding a SwiftUI implementation now. Maybe just as an “add on” to the Swift version, but that’s a different discussion anyway...

**Comment by iDevelopper** [comment]
First, I think we could expose the swift framework to Objective-C. What about this?

**Comment by vtourraine** [comment]
Using `@objc` annotations? That seems like a good idea (I just hope we’re not overlooking some kind of obscure feature or optimization only available in Swift-only libraries). Feel free to open a pull request on AcknowList (or at least an Issue to call this out).

**Comment by iDevelopper** [comment]
I don't understand why but it already works without using @objC annotations!

[AcknowListTestObjC.zip](https://github.com/vtourraine/VTAcknowledgementsViewController/files/5916494/AcknowListTestObjC.zip)

**Comment by vtourraine** [comment]
Interesting. Maybe because they‘re subclasses of Objective-C classes, so there’s some inherited compatibility?

But still, only the `[[class alloc] init]` initializer is available. And objects like `Acknow` and the parser are not available at all. So there’s still work to be done to have AcknowList really Objective-C compatible.

**Comment by iDevelopper** [comment]
Exact, you are right! Ouf, because I did not understand!
Not all functions need to be compatible? Do you need the parser to be public? Or internal?

**Comment by vtourraine** [comment]
Everything that is open/public in Swift should also be accessible from Objective-C, if we want it to be fully compatible.

**Comment by iDevelopper** [comment]
I don't think so. We need to expose only the functions that a user will use in an Objective-C project.

**Comment by vtourraine** [comment]
I understand your point of view, but some developers will want to have access to everything that is already accessible in Swift, especially if we want to retire the Objective-C library later on.
